### PR TITLE
test(e2e-components): rewrite all e2e sheet components to accomplish vue3 

### PR DESCRIPTION
**Pull request by joseacabaneros**
Rewrite all cypress component tests to accomplish vue3 requirements.
Passing CI ✅

### CODE REVIEW

File: packages/x-components/cypress.config.ts [hunk 1]
```diff
@@ -9,34 +7,41 @@ export default defineConfig({
...
-      await addCucumberPreprocessorPlugin(on, config);
-      on(
-        'file:preprocessor',
-        require('@bahmutov/cypress-esbuild-preprocessor')({
-          plugins: [createEsbuildPlugin(config)]
-        })
+      const { createEsbuildPlugin } = require('@badeball/cypress-cucumber-preprocessor/esbuild');
+      const createBundler = require('@bahmutov/cypress-esbuild-preprocessor');
+
+      // await here
```
Comment (victorcg88) [inline packages/x-components/cypress.config.ts:25]: Maybe we should improve this comment.

File: packages/x-components/tests/unit/banner.spec.ts [hunk 2]
```diff
@@ -60,13 +33,8 @@ describe('testing Banner component', () => {
...
 
   it('banner is not rendered if image load fails', () => {
-    const { getBanner, getBannerImage } = mountBanner({
-      banner: createBannerStub('banner', {
-        title: 'Search UIs',
-        url: 'https://empathy.co',
-        position: 1
-      })
-    });
+    const { getBanner, getBannerImage } = render(createBannerStub('banner', bannerBase));
```
Comment (victorcg88) [inline packages/x-components/tests/unit/banner.spec.ts:36]:
```suggestion
    const { getBanner, getBannerImage } = render();
```

We could make `createBannerStub('banner', bannerBase)` the default value for the `banner` arg of `render`. WDYT? I know it is not a big deal here, but in any case, I think it is a good pattern to follow.``

**Review by victorcg88 — COMMENTED** [review]
Mega job! 🔝 

image

File: packages/x-components/cypress.config.ts (hunk 1 above)
Comment (joseacabaneros) [reply packages/x-components/cypress.config.ts]: Yep. I rewrote this config following the official doc for cucumber

File: packages/x-components/tests/unit/banner.spec.ts (hunk 2 above)
Comment (joseacabaneros) [reply packages/x-components/tests/unit/banner.spec.ts]: I like it! Reformatted for default param value

**Review by victorcg88 — APPROVED** [review]

**Pull request merged**